src/Container/DefinitionResolver.php

```diff
@@ -96,7 +96,9 @@ public function resolve(DefinitionInterface $definition)
      */
     private function resolveReference($value)
     {
-        if ($value instanceof ReferenceInterface) {
+        if (is_array($value)) {
+            return array_map([$this, 'resolveReference'], $value);
```

Comment:
Maybe best to do this recursively, otherwise we still have the issue with a reference in an array in an array. Like, with `array_walk_recursive` or something:

```php
array_walk_recursive($value, function(&$v) {
    $v = $this->resolveReference($v);
});
```